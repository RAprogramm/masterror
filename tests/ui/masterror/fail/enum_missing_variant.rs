use masterror::{AppCode, AppErrorKind, Masterror};

#[derive(Debug, Masterror)]
enum Mixed {
    #[error("with")]
    #[masterror(code = AppCode::Internal, category = AppErrorKind::Internal)]
    With,
    #[error("missing")]
    Missing
}

fn main() {}
