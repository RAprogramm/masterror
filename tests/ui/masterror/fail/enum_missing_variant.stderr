error: all variants must use #[masterror(...)] to derive masterror::Error conversion
 --> tests/ui/masterror/fail/enum_missing_variant.rs:8:5
  |
8 | /     #[error("missing")]
9 | |     Missing
  | |___________^

warning: unused imports: `AppCode` and `AppErrorKind`
 --> tests/ui/masterror/fail/enum_missing_variant.rs:1:17
  |
1 | use masterror::{AppCode, AppErrorKind, Masterror};
  |                 ^^^^^^^  ^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` (part of `#[warn(unused)]`) on by default
